### ISSUE TITLE
Classic pdflatex ENOENT 

### ISSUE THREAD

**Issue by Pedro-Roque** [James-Yu/LaTeX-Workshop#3032]
### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: **Yes**

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: **Yes**

### If your issue is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.

You can compile a TeX document manually?: **Yes**

## Describe the bug [Required]

I can compile the latex document using `pdflatex --shell-escape main.tex` without any issues. My $PATH contains the directory to the TeX executables (they reside in /usr/bin, and my PATH is: `$PATH: /home/roque/.poetry/bin:/opt/ros/noetic/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin`. However, I still get the error in the log.

### To Reproduce

1. Open a tex file
2. Save the .tex file to trigger compilation

### Expected behavior

Expected to compile the .tex file.

## Logs [Required]
[22:24:40] Initializing LaTeX Workshop.
[22:24:40] Extension root: /home/roque/.vscode/extensions/james-yu.latex-workshop-8.23.0
[22:24:40] $PATH: /home/roque/.poetry/bin:/opt/ros/noetic/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[22:24:40] $SHELL: /usr/bin/zsh
[22:24:40] vscode.env.appName: Visual Studio Code
[22:24:40] vscode.env.remoteName: undefined
[22:24:40] vscode.env.uiKind: 1
[22:24:40] editor.acceptSuggestionOnEnter: "on"
[22:24:40] latex-workshop.bind.enter.key: true
[22:24:40] latex-workshop.docker.enabled: false
[22:24:40] latex-workshop.docker.image.latex: ""
[22:24:40] latex-workshop.hover.preview.mathjax.extensions: []
[22:24:40] latex-workshop.intellisense.package.enabled: true
[22:24:40] latex-workshop.intellisense.update.aggressive.enabled: false
[22:24:40] latex-workshop.intellisense.update.delay: 1000
[22:24:40] latex-workshop.latex.autoBuild.run: "onFileChange"
[22:24:40] latex-workshop.latex.outDir: "%DIR%"
[22:24:40] latex-workshop.latex.recipes: [
 {
  "name": "latexmk 🔃",
  "tools": [
   "latexmk"
  ]
 },
 {
  "name": "latexmk (latexmkrc)",
  "tools": [
   "latexmk_rconly"
  ]
 },
 {
  "name": "latexmk (lualatex)",
  "tools": [
   "lualatexmk"
  ]
 },
 {
  "name": "pdflatex ➞ bibtex ➞ pdflatex × 2",
  "tools": [
   "pdflatex",
   "bibtex",
   "pdflatex",
   "pdflatex"
  ]
 },
 {
  "name": "Compile Rnw files",
  "tools": [
   "rnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Jnw files",
  "tools": [
   "jnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "tectonic",
  "tools": [
   "tectonic"
  ]
 }
]
[22:24:40] latex-workshop.latex.tools: [
 {
  "name": "latexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-pdf",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "lualatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-lualatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "latexmk_rconly",
  "command": "latexmk",
  "args": [
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "pdflatex",
  "command": "pdflatex",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "bibtex",
  "command": "bibtex",
  "args": [
   "%DOCFILE%"
  ],
  "env": {}
 },
 {
  "name": "rnw2tex",
  "command": "Rscript",
  "args": [
   "-e",
   "knitr::opts_knit$set(concordance = TRUE); knitr::knit('%DOCFILE_EXT%')"
  ],
  "env": {}
 },
 {
  "name": "jnw2tex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"tex\")"
  ],
  "env": {}
 },
 {
  "name": "jnw2texmintex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"texminted\")"
  ],
  "env": {}
 },
 {
  "name": "tectonic",
  "command": "tectonic",
  "args": [
   "--synctex",
   "--keep-logs",
   "%DOC%.tex"
  ],
  "env": {}
 }
]
[22:24:40] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
[22:24:40] Creating a new file watcher.
[22:24:40] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[22:24:40] Creating PDF file watcher.
[22:24:40] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
[22:24:40] Creating Bib file watcher.
[22:24:40] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[22:24:40] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[22:24:40] [Server] Creating LaTeX Workshop http and websocket server.
[22:24:40] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[22:24:40] LaTeX Workshop initialized.
[22:24:40] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[22:24:40] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[22:24:40] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[22:24:40] Current workspace folders: ["file:///home/roque/space_ws/src/VisCoor/paper/hello"]
[22:24:40] Current workspaceRootDir: file:///home/roque/space_ws/src/VisCoor/paper/hello
[22:24:40] Found root file from active editor: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] Root file changed: from undefined to /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] Start to find all dependencies.
[22:24:40] Root file languageId: latex
[22:24:40] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":45493}
[22:24:40] Reset file watcher.
[22:24:40] Parsing a file and its subfiles: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] Parse fls file.
[22:24:40] Cannot find fls file: /home/roque/space_ws/src/VisCoor/paper/hello/hello.fls
[22:24:40] Added to file watcher: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] Snippet data loaded.
[22:24:40] Checking for duplicate labels: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex.
[22:24:40] onDidSaveTextDocument triggered: file:///home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] File watcher - file changed: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] Parsing a file and its subfiles: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] Checking for duplicate labels: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex.
[22:24:40] Auto build started detecting the change of a file: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] BUILD command invoked.
[22:24:40] The document of the active editor: file:///home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:40] The languageId of the document: latex
[22:24:40] Build using the external command: pdflatex --shell-escape main.tex 
[22:24:40] cwd: /home/roque/space_ws/src/VisCoor/paper/hello
[22:24:40] External build process spawned. PID: undefined.
[22:24:40] Build fatal error: spawn pdflatex --shell-escape main.tex ENOENT, . PID: undefined. Does the executable exist?
[22:24:42] Manager.fileWatcher.getWatched: {"/home/roque/space_ws/src/VisCoor/paper/hello":["hello.tex"]}
[22:24:42] Manager.filesWatched: ["/home/roque/space_ws/src/VisCoor/paper/hello/hello.tex"]
[22:24:42] BibWatcher.bibWatcher.getWatched: {}
[22:24:42] BibWatcher.bibsWatched: []
[22:24:42] PdfWatcher.pdfWatcher.getWatched: {}
[22:24:42] PdfWatcher.pdfsWatched: []
[22:24:48] Current workspace folders: ["file:///home/roque/space_ws/src/VisCoor/paper/hello"]
[22:24:48] Current workspaceRootDir: file:///home/roque/space_ws/src/VisCoor/paper/hello
[22:24:48] Found root file from active editor: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:48] Keep using the same root file: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:49] onDidSaveTextDocument triggered: file:///home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:49] File watcher - file changed: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:49] Parsing a file and its subfiles: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:49] Checking for duplicate labels: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex.
[22:24:49] Auto build started detecting the change of a file: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:49] BUILD command invoked.
[22:24:49] The document of the active editor: file:///home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:49] The languageId of the document: latex
[22:24:49] Build using the external command: pdflatex --shell-escape main.tex 
[22:24:49] cwd: /home/roque/space_ws/src/VisCoor/paper/hello
[22:24:49] External build process spawned. PID: undefined.
[22:24:49] Build fatal error: spawn pdflatex --shell-escape main.tex ENOENT, . PID: undefined. Does the executable exist?
[22:24:50] Manager.fileWatcher.getWatched: {"/home/roque/space_ws/src/VisCoor/paper/hello":["hello.tex"]}
[22:24:50] Manager.filesWatched: ["/home/roque/space_ws/src/VisCoor/paper/hello/hello.tex"]
[22:24:50] BibWatcher.bibWatcher.getWatched: {}
[22:24:50] BibWatcher.bibsWatched: []
[22:24:50] PdfWatcher.pdfWatcher.getWatched: {}
[22:24:50] PdfWatcher.pdfsWatched: []
[22:24:51] Current workspace folders: ["file:///home/roque/space_ws/src/VisCoor/paper/hello"]
[22:24:51] Current workspaceRootDir: file:///home/roque/space_ws/src/VisCoor/paper/hello
[22:24:51] Found root file from active editor: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex
[22:24:51] Keep using the same root file: /home/roque/space_ws/src/VisCoor/paper/hello/hello.tex

**EDIT:** using the file as main.tex also doesn't work

### Developer Tools Console [Required]

onDidChangeNotification @ notificationAlerts.ts:42  (Related to the error above)

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: Ubuntu 20.04
 - VS Code version: 1.63.2
 - Extension version: 8.23.0
 - TeX distribution version: TeX Live 2019/Debian

### Are you using VSCodium?

No

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

No

**Comment by tamuratak** [comment]
Your configuration is broken. You are setting the whole string `pdflatex --shell-escape main.tex` as a command.

See https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#external-build-command

**Comment by jlelong** [comment]
https://github.com/James-Yu/LaTeX-Workshop/blob/c2d51873d2313daf227276fe9de53baea655ee2c/src/components/builder.ts#L129

@tamuratak I suggest we improve logging commands by clearly marking the command and every argument. We could use the comma as a separator

```ts
this.extension.logger.addLogMessage(`Build using the external command: ${command}, ${args.length > 0 ? args.join(', '): ''}`)
```

**Comment by tamuratak** [comment]
I will merge the following commit when I have time.

- https://github.com/tamuratak/LaTeX-Workshop/commit/cf8ba0e1894f437974aadd332d53ed6722038e77
